src/Microsoft.AspNet.Authentication/AuthenticationHandler.cs

```diff
@@ -61,12 +64,13 @@ internal AuthenticationOptions BaseOptions
 
         public bool Faulted { get; set; }
 
-        protected async Task BaseInitializeAsync([NotNull] AuthenticationOptions options, [NotNull] HttpContext context, [NotNull] ILogger logger)
+        protected async Task BaseInitializeAsync([NotNull] AuthenticationOptions options, [NotNull] HttpContext context, [NotNull] ILogger logger, [NotNull] IUrlEncoder encoder)
         {
             _baseOptions = options;
             Context = context;
             RequestPathBase = Request.PathBase;
             Logger = logger;
+            UrlEncoder = encoder;
```

Comment:
Another option we could do, is remove the ability to configure options as part of Use, so force an explicitly AddCookie/Auth for each instance of middleware, the downside of this is with multiple instances of the same middleware, but I guess the overload would be UseXyzAuthentication(string authenticationScheme) and they would just be required to specify that correctly as part of the ConfigureXyzAuthentication(() => { }, authenticationScheme